src/Sulu/Component/Content/Types/ResourceLocator/Strategy/ResourceLocatorStrategy.php

```diff
@@ -129,21 +121,21 @@ public function generate($title, $parentUuid, $webspaceKey, $languageCode, $segm
         return $path;
     }
 
-    /**
-     * Returns the first ancestor-or-self of the given document which is published and therefore has an
-     * assigned resource locator. If all ancestor documents are unpublished, the root document is returned.
-     *
-     * @param object $document
-     *
-     * @return object
-     */
-    private function getPublishedAncestorOrSelf($document)
+    public function getClosestResourceLocator($uuid, $webspaceKey, $languageCode)
     {
-        while (!$document->getPublished() && $document instanceof ParentBehavior) {
-            $document = $document->getParent();
+        if (!$uuid) {
+            return null;
         }
 
-        return $document;
+        $document = $this->documentManager->find($uuid, $languageCode, ['load_ghost_content' => false]);
+
+        do {
+            try {
+                return $this->loadByContentUuid($this->documentInspector->getUuid($document), $webspaceKey, $languageCode);
+            } catch (ResourceLocatorNotFoundException $e) {
+                $document = $document->getParent();
+            }
+        } while ($document instanceof ParentBehavior);
```

Comment:
Because it is not necessary anymore. We now check in the actual tree if there is a URL. Using `getPublished` was more or less an assumption, which turned out not to be always true.